app_grim_reefer/src/main.c

```diff
@@ -27,54 +47,168 @@ const struct gpio_dt_spec led2 = GPIO_DT_SPEC_GET(LED2_NODE, gpios);
 const struct gpio_dt_spec ldo_enable = GPIO_DT_SPEC_GET(LDO_EN_NODE, gpios);
 const struct gpio_dt_spec cam_enable = GPIO_DT_SPEC_GET(CAM_EN_NODE, gpios);
 
-static int init(void) {
+#define BUZZER_NODE DT_NODELABEL(buzzer)
+const struct gpio_dt_spec buzzer = GPIO_DT_SPEC_GET(BUZZER_NODE, gpios);
+
+#define DBG_SERIAL_NODE DT_ALIAS(debug_serial)
+const struct device *const debug_serial_dev = DEVICE_DT_GET(DBG_SERIAL_NODE);
+
+#define BME280_NODE DT_NODELABEL(bme280)
+const struct device *bme280_dev = DEVICE_DT_GET(BME280_NODE);
+
+#define LSM6DSL_NODE DT_NODELABEL(lsm6dsl)
+const struct device *lsm6dsl_dev = DEVICE_DT_GET(LSM6DSL_NODE);
+
+#define FLASH_NODE DT_NODELABEL(w25q512)
+const struct device *flash_dev = DEVICE_DT_GET(FLASH_NODE);
+
+// Inas
+#define INA_BAT_NODE DT_NODELABEL(ina260_battery)
+const struct device *ina_bat_dev = DEVICE_DT_GET(INA_BAT_NODE);
+
+#define INA_LDO_NODE DT_NODELABEL(ina260_ldo)
+const struct device *ina_ldo_dev = DEVICE_DT_GET(INA_LDO_NODE);
+
+#define INA_GRIM_NODE DT_NODELABEL(ina260_3v3)
+const struct device *ina_grim_dev = DEVICE_DT_GET(INA_GRIM_NODE);
+
+static const struct adc_dt_spec adc_chan0 = ADC_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), 0);
```

Comment:
NIT: Declare devices locally where they are needed instead of globals. Actively try to avoid globals if possible. This will be especially the case if we move to C++.

Reply:
Idrk what the best way to go abt this is where devices are needed many places and I think its reasonable to do the startup all in one place. My thought was that you get your devices, make sure theyre functioning, then fan them out to where they need to be. For example, the LDO pin doesnt really have an 'owner' so to speak since its needed for adc control and buzzer control. If they were setup locally both the data fetching part and the buzzer part would have to repeat init or have to control initialization order across threads both of which seems messier to me. Do you or @Naquino14 have guidance on this?